exchanges/order/orders.go

```diff
@@ -105,6 +113,27 @@ func (s *Submit) Validate(opt ...validate.Checker) error {
 		return ErrPriceMustBeSetIfLimitOrder
 	}
 
+	requirements, err := exch.GetTradingRequirements()
+	if err != nil {
+		return err
+	}
+
+	if requirements.ClientOrderID && s.ClientOrderID == "" {
+		return fmt.Errorf("submit validation error %w, client order ID must be set to satisfy submission requirements", ErrClientOrderIDMustBeSet)
+	}
+
+	if s.Type == Market && s.AssetType == asset.Spot && s.Side.IsLong() {
+		if requirements.SpotMarketOrderAmountPurchaseQuotationOnly && s.QuoteAmount == 0 {
+			return fmt.Errorf("submit validation error %w, quote amount to be sold must be set to 'QuoteAmount' field to satisfy trading requirements", ErrAmountMustBeSet)
+		}
+	}
```

Comment:
I think you should lead with the requirement:
```go
if requirements.SpotMarketOrderAmountPurchaseQuotationOnly && s.QuoteAmount == 0 && s.Type == Market && s.AssetType == asset.Spot && s.Side.IsLong() {
		return fmt.Errorf("submit validation error %w, quote amount to be sold must be set to 'QuoteAmount' field to satisfy trading requirements", ErrAmountMustBeSet)
	}
```
Makes it easier to follow how its validating. Same for below